calculator.js

```diff
@@ -0,0 +1,47 @@
+const operators = ['+', '-', '*', '/'];
+
+exports.create_calculator = function (operator) {
+  const numbers = [];
+
+  return {
+    add_number: function (number) {
+      numbers.push(number);
+    },
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
+      if (numbers.length === 0) {
+        throw 'Cannot calculate a result because no numbers were given';
+      }
+
+      let answer = numbers[0];
+
+      let i = 0;
+      numbers.forEach(function (number) {
+        if (i === 0) {
+          i++;
+          return;
+        }
+
+        switch (operator) {
+          case '+':
+            answer = answer + number;
+            break;
+          case '-':
+            answer = answer - number;
+            break;
+          case '/':
+            answer = answer / number;
+            break;
+          default:
+            console.log("Invalid operator " + operator);
```

Comment:
Should we throw this error when creating the calculator object, rather than during execution of the calculation. This way any errors regarding the operator will be thrown when the calculator is created, so it should be easier to spot when an invalid operator is used! ☺️

One way to do this would be to add in the constructor:
```
if (!operators.includes(operator)) {
  // throw an error here
}
```

Reply:
On a related note, should this be throwing rather than logging? We should be consistent with the above error for when there are no numbers to calculate 🙂